### PR TITLE
Add support for users and basic jwt authorization

### PR DESCRIPTION
**Pull request by bewee**

### CODE REVIEW

File: src/router/login_router.rs [hunk 1]
```diff
@@ -1,24 +1,44 @@
...
+        {
+            return Err(status::Custom(
+                Status::Unauthorized,
+                "Unauthorized".to_owned(),
+            ));
+        }
+
+        let jwt = jwt::issue_token(user.id)
+            .await
+            .to_rocket("Failed to issue token", Status::BadRequest)?;
```
Comment (tim-hellhake) [inline src/router/login_router.rs:36]: This looks more like a server error than a client error.
Maybe `InternalServerError`?

File: src/router/login_router.rs [hunk 2]
```diff
@@ -1,24 +1,44 @@
...
-}
-
 #[post("/", data = "<data>")]
-fn login(data: Json<Login>) -> Json<Jwt> {
-    Json(Jwt {
-        jwt: format!("{}:{}", data.email, data.password),
-    })
+async fn login(data: Json<Login>) -> Result<Json<Jwt>, status::Custom<String>> {
+    let user = call!(Db.GetUser::ByEmail(data.0.email))
+        .to_rocket("Failed to get user", Status::BadRequest)?;
```
Comment (tim-hellhake) [inline src/router/login_router.rs:23]: This looks more like a server error than a client error.
Maybe `InternalServerError`?

**Review by tim-hellhake — CHANGES_REQUESTED** [review]

File: src/router/login_router.rs (hunk 2 above)
Comment (bewee) [reply src/router/login_router.rs:23]: Agree. However, this flaw exists all over the code base. Therefore, I would prefer to solve it in a separate PR

File: src/router/login_router.rs (hunk 1 above)
Comment (bewee) [reply src/router/login_router.rs:36]: Same here

File: src/router/login_router.rs (hunk 2 above)
Comment (tim-hellhake) [reply src/router/login_router.rs:23]: Agreed

File: src/router/login_router.rs (hunk 1 above)
Comment (tim-hellhake) [reply src/router/login_router.rs:36]: Agreed

**Review by tim-hellhake — APPROVED** [review]

**Review by tim-hellhake — APPROVED** [review]

**Pull request merged**